source/common/quic/envoy_quic_client_session.cc

```diff
@@ -13,15 +15,40 @@
 namespace Envoy {
 namespace Quic {
 
+// Quic connection doesn't use transport socket object to do handshake. This callback
+// implementation is used by the cert verifier to access Network::Connection interfaces.
+class ConnectionWrapper : public Network::TransportSocketCallbacks {
```

Comment:
If this class is intended to be used for cert validation context, how about naming it something like "CertValidationContext"?

Reply:
done